poradnia/users/views.py

```diff
@@ -11,11 +12,17 @@
 from .utils import PermissionMixin
 
 
-class UserDetailView(PermissionMixin, DetailView):
+class UserDetailView(PermissionRequiredMixin, DetailView):
     model = User
-    # These next two lines tell the view to index lookups by username
     slug_field = "username"
     slug_url_kwarg = "username"
+    permission_required = "users.can_view_other"
+    raise_exception = True
+
+    def has_permission(self, *args, **kwargs):
+        if self.kwargs['username'] == self.request.user.username:
+            return True
+        return super(UserDetailView, self).has_permission(*args, **kwargs)
```

Comment:
Może użycie ```super ``` to nadużycie, ale takim zapisem podkreślić chciałem, że tylko w tym zakresie robię wyjątek.